### PR TITLE
Compression

### PR DESCRIPTION
**Pull request by chonma0ctopus**
We implemented functions to combine a large number of weak absorption lines that form a continuum level into a smaller number of "pseudo lines".
Lines with similar wavelength, elower, and gamma are gathered, except for relatively strong absorption lines at the expected temperatures.

### CODE REVIEW

File: src/exojax/utils/exofile.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+"""
+import pathlib
+import vaex
+import json
+
+def save_exomol_mif(output,nu_grid,cnu,indexnu,logsij0,A,alpha_ref,n_Texp,elower,extension=".hdf5"):
+    """Save inputs as MODIT Input File Format for ExoMol (emif)
+
+    Args:
```
Comment (HajimeKawahara) [inline src/exojax/utils/exofile.py:16]: Can you give Args?

File: src/exojax/spec/plg.py [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+    #frozen criterion
+    num_unique=np.bincount(eindex,minlength=Ng) # number of the lines in a bin
+    
+    lmask=(num_unique>=Ncrit)
+    erange=range(0,Ng)
+    frozen_eindex=np.array(erange)[lmask]
+    frozen_mask=np.isin(eindex,frozen_eindex)
+    
+    #Retake relatively strong lines.
+    SijTgue_frozen = SijT(Tuplim+500., \
```
Comment (HajimeKawahara) [inline src/exojax/spec/plg.py:125]: Why 500? Should we use a variable?

File: src/exojax/spec/plg.py [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+                    qT=mdb.qr_interp(Tuplim+500.))
+    persist_freezing = SijTgue_frozen < max(SijTgue_frozen) / threshold_persist_freezing
+    index_persist_freezing = np.where(frozen_mask)[0][persist_freezing]
+    frozen_mask = np.isin(np.arange(len(frozen_mask)), index_persist_freezing)
+        
+    #qlogsij0
+    Sij=np.exp(mdb.logsij0)
+    qlogsij0=np.bincount(eindex,weights=Sij*(1.0-cont)*frozen_mask,minlength=Ng)
+    qlogsij0=qlogsij0+np.bincount(eindex+1,weights=Sij*cont*frozen_mask,minlength=Ng)
+    with warnings.catch_warnings():
```
Comment (HajimeKawahara) [inline src/exojax/spec/plg.py:136]: Just comment: cool. I should have used `warnings` package in the code I wrote. I will do that in the forthcoming refactoring

File: src/exojax/spec/plg.py [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+       frozen_mask: mask for frozen lines into pseudo lines 
+       nonzeropl_mask: mask for pseudo-lines w/ non-zero
+
+    """
+    elower = mdb.elower
+    Nnugrid=len(nu_grid)
+    Tref = 296.0
+    Tuplim = Tgue + errTgue
+    Tuplimc = Tuplim * coefTgue
+    preov = max(- hcperk*(elower/Tuplimc - elower/Tref)) - 80. if preov==0. else preov
```
Comment (HajimeKawahara) [inline src/exojax/spec/plg.py:47]: What's 80? should we parametrize this?

**Comment by HajimeKawahara** [comment]
Thanks a lot! @chonma0ctopus I confirmed that plg_methane_test.py and plg_h2o_test.py passed the test.
I will make docs based on the tutorials you made, but I feel more explanations are needed. Can you add more explanations on the tutorials?

**Comment by chonma0ctopus** [comment]
Thank you for the review and docs!

> Can you add more explanations on the tutorials?
Certainly, I'll do it.

File: src/exojax/utils/exofile.py (hunk 1 above)
Comment (chonma0ctopus) [reply src/exojax/utils/exofile.py:16]: I'm afraid that this file is not made by me. 
So I didn't modify it at all. Shall I get it organized?

Comment (HajimeKawahara) [reply src/exojax/utils/exofile.py:16]: Ah OK, sorry. If you do not need this function, remove it.

**Comment by chonma0ctopus** [comment]
Thank you for your patience! @HajimeKawahara
All your review comments above were incorporated and also unittests have been added.
(Note that I pulled your premodit branch, so it is better to merge it into develop after premodit.)

**Comment by HajimeKawahara** [comment]
@chonma0ctopus Thanks! 
In my environment, it took ~ 30 sec to run a unit test `optimize_coefTgue_test.py`.  It's too long for the unit test. Can you reduce the computation time?

**Comment by chonma0ctopus** [comment]
That's true. It is taking too long because the scipy.optimize.least_squares is used in the function "optimize_coefTgue" that I intended to test in this file.
How about avoiding testing optimize_coefTgue and instead, only test the function "diff_frozen_vs_pseudo" that is running inside the "optimize_coefTgue"?

**Comment by HajimeKawahara** [comment]
confirmed that the time was improved. Thanks! 

>(Note that I pulled your premodit branch, so it is better to merge it into develop after premodit.)
OK. I will merge this branch after the merge of `premodit` to `develop`.

**Comment by HajimeKawahara** [comment]
Confirmed that the reverse test worked.

**Pull request merged**